packages/peregrine/lib/talons/MegaMenu/useMegaMenu.js

```diff
@@ -91,7 +91,7 @@ export const useMegaMenu = (props = {}) => {
     );
 
     const megaMenuData = useMemo(() => {
-        return data ? processData(data.categoryList[0]) : null;
+        return data ? processData(data.categoryList[0]) : {};
```

Comment:
Generally we don't want to return an entity when the data is not yet returned - we can use the lack of it, and the `loading` state from the query, to know when to render a loading state.

For example:

```
if loading && !data, render loading spinner
if loading && data (cached/stale), render data
```

As always, loading spinner/states are debatable, and I'm not actually sure what loading state you have implemented in this feature. I just wanted to drop some info :)

Reply:
In this case, since the "loading" state is just rendering nothing, we can/should leave this :) I have simplified the export below since you already did the assignment here.